AEWeb: display the size when listing the files of a directory

### Is your feature request related to a problem?

_No response_

### Describe the solution you'd like

We added the size (in bytes) of every files in the reference transaction. 
We can now add a column to display the size of every files when listing the files of a directory: 

![image](https://user-images.githubusercontent.com/74045243/210397946-5cf43eba-8770-4a38-b0b6-e5fef049e286.png)


### Additional context

Example of a reference transaction:

```
{
  "aewebVersion": 1,
  "hashFunction": "sha1",
  "metaData": {
    "dir1/file10.txt": {
      "hash": "d5772baf70d86e28b3349580569f58b3ec1a8db9",
      "size": 8,
      "encoding": "gzip",
      "addresses": [
        "0000942ea8e9246f51f589373075bf09e3490c34138ce63702b3a27336bac8316168"
      ]
    },
    ...
}
```

src/c/crypto/tpm/lib.c:5:10: fatal error: tss2/tss2_esys.h: No such file or directory
    5 | #include <tss2/tss2_esys.h>
      |          ^~~~~~~~~~~~~~~~~~

> src/c/crypto/tpm/lib.c:5:10: fatal error: tss2/tss2_esys.h: No such file or directory 5 | #include <tss2/tss2_esys.h> | ^~~~~~~~~~~~~~~~~~

If you run this manually, do you have any error ? 
```bash
wget https://github.com/tpm2-software/tpm2-tss/releases/download/3.2.1-rc2/tpm2-tss-3.2.1-rc2.tar.gz
tar -xf tpm2-tss-3.2.1-rc2.tar.gz --one-top-level=tpm2-tss --strip-components 1
rm tpm2-tss-3.2.1-rc2.tar.gz && cd tpm2-tss
./configure --with-udevrulesdir=/etc/udev/rules.d
make -j$(nproc)

sudo make install
sudo sed -i "s/tss/$(whoami)/gi" /etc/udev/rules.d/tpm-udev.rules
sudo udevadm control --reload-rules && sudo udevadm trigger
sudo ldconfig
sudo apt install tpm2-tools -y
```

--- a/lib/archethic_web/controllers/api/web_hosting_controller/directory_listing.ex
+++ b/lib/archethic_web/controllers/api/web_hosting_controller/directory_listing.ex
@@ -68,10 +68,10 @@ defmodule ArchethicWeb.API.WebHostingController.DirectoryListing do
 
     json_content_subset
     |> Enum.map(fn
-      {key, %{"addresses" => address}} ->
+      {key, %{"addresses" => address, "size" => size}} ->
         case Path.split(key) do
           [^key] ->
-            {:file, key, address}
+            {:file, key, address, size}
 
           [dir | _] ->
             {:dir, dir}
@@ -80,25 +80,26 @@ defmodule ArchethicWeb.API.WebHostingController.DirectoryListing do
     |> Enum.uniq()
     # sort directory last, then DESC order (it will be accumulated in reverse order below)
     |> Enum.sort(fn
-      {:file, a, _}, {:file, b, _} ->
+      {:file, a, _, _}, {:file, b, _, _} ->
         a > b
 
       {:dir, a}, {:dir, b} ->
         a > b
 
-      {:file, _, _}, {:dir, _} ->
+      {:file, _, _, _}, {:dir, _} ->
         true
 
-      {:dir, _}, {:file, _, _} ->
+      {:dir, _}, {:file, _, _, _} ->
         false
     end)
     |> Enum.reduce(%{dirs: [], files: []}, fn
-      {:file, name, addresses}, %{dirs: dirs_acc, files: files_acc} ->
+      {:file, name, addresses, size}, %{dirs: dirs_acc, files: files_acc} ->
         item = %{
           href: %{href: Path.join(request_path, name)},
           last_modified: timestamp,
           addresses: addresses,
-          name: name
+          name: name,
+          size: size
         }
 
         %{dirs: dirs_acc, files: [item | files_acc]}

--- a/lib/archethic_web/templates/dir_listing/index.html.heex
+++ b/lib/archethic_web/templates/dir_listing/index.html.heex
@@ -137,6 +137,7 @@
                             </tbody>
                         </table>
                     </td>
+                    <td class="size"></td>
                     <td class="transaction"></td>
                     <td class="last-modified"><%= format_date(dir.last_modified) %></td>
                 </tr>

--- a/lib/archethic_web/templates/dir_listing/index.html.heex
+++ b/lib/archethic_web/templates/dir_listing/index.html.heex
@@ -116,6 +116,7 @@
         <thead>
             <tr>
                 <th>Name</th>
+                <th class="size">Size(Bytes)</th>
                 <th class="transaction">Transactions</th>
                 <th class="last-modified">Last Modified</th>
             </tr>
@@ -147,6 +148,7 @@
                             </tbody>
                         </table>
                     </td>
+                    <td class="size"><%= file.size %></td>
                     <td class="transaction">
                     <%= for address <- prepare_addresses(file.addresses) do %>
                         <a class="address comma"{address.href}><%= address.text %></a>

--- a/lib/archethic_web/templates/dir_listing/index.html.heex
+++ b/lib/archethic_web/templates/dir_listing/index.html.heex
@@ -47,6 +47,10 @@
     .comma:not(:empty) ~ .comma:not(:empty):before {
         content: ", ";
     }
+    table.index .size{
+      width: 1px;
+      white-space: nowrap;
+    }
 
     i.icon {
         width: 16px;

--- a/test/archethic_web/controllers/api/web_hosting_controller_test.exs
+++ b/test/archethic_web/controllers/api/web_hosting_controller_test.exs
@@ -409,42 +409,49 @@ defmodule ArchethicWeb.API.WebHostingControllerTest do
       "hashFunction": "sha-1",
       "metaData":{
         "dir1/file10.txt":{
+          "size": 10,
           "encoding":"gzip",
           "addresses":[
             "000071fbc2205f3eba39d310baf15bd89a019b0929be76b7864852cb68c9cd6502de"
           ]
         },
         "dir1/file11.txt":{
+          "size": 10,
           "encoding":"gzip",
           "addresses":[
             "000071fbc2205f3eba39d310baf15bd89a019b0929be76b7864852cb68c9cd6502de"
           ]
         },
         "dir2/hello.txt":{
+          "size": 10,
           "encoding":"gzip",
           "addresses":[
             "000071fbc2205f3eba39d310baf15bd89a019b0929be76b7864852cb68c9cd6502de"
           ]
         },
         "dir3/index.html":{
+          "size": 10,
           "encoding":"gzip",
           "addresses":[
             "000071fbc2205f3eba39d310baf15bd89a019b0929be76b7864852cb68c9cd6502de"
           ]
         },
         "file1.txt":{
+          "size": 10,
           "encoding":"gzip",
           "addresses":[
             "000071fbc2205f3eba39d310baf15bd89a019b0929be76b7864852cb68c9cd6502de"
           ]
         },
         "file2.txt":{
+          "size": 10,
           "encoding":"gzip",
           "addresses":[
             "000071fbc2205f3eba39d310baf15bd89a019b0929be76b7864852cb68c9cd6502de"
           ]
         },
         "file3.txt":{
+          "size": 10,
           "encoding":"gzip",
           "addresses":[
             "000071fbc2205f3eba39d310baf15bd89a019b0929be76b7864852cb68c9cd6502de"
@@ -511,12 +518,14 @@ defmodule ArchethicWeb.API.WebHostingControllerTest do
       "hashFunction": "sha-1",
       "metaData":{
         "address_content.png":{
+          "size": 20,
           "encoding":"gzip",
           "addresses":[
             "000071fbc2205f3eba39d310baf15bd89a019b0929be76b7864852cb68c9cd6502de"
           ]
         },
         "concat_content.png":{
+          "size": 30,
           "encoding":"gzip",
           "addresses":[
             "000071fbc2205f3eba39d310baf15bd89a019b0929be76b7864852cb68c9cd6502de",
@@ -622,6 +631,7 @@ defmodule ArchethicWeb.API.WebHostingControllerTest do
       "hashFunction": "sha-1",
       "metaData":{
         "folder/hello_world.html":{
+          "size": 20,
           "encoding":"base64",
           "addresses":[
             "000071fbc2205f3eba39d310baf15bd89a019b0929be76b7864852cb68c9cd6502de"

